### PR TITLE
Add the new eCBIS flavor | main

### PR DESCRIPTION
**Pull request by dubdabasoduba**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

- This addition will allow our client to install apps targetting the preview server and the prod server on the same device. 
- The logos on the preview APK have been changed for clarity on what each APK does

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/quest/build.gradle.kts [hunk 1]
```diff
@@ -171,6 +171,13 @@ android {
...
       applicationIdSuffix = ".ecbis"
       versionNameSuffix = "-ecbis"
+      manifestPlaceholders["appLabel"] = "MOH eCBIS Preview"
+    }
+
+    create("ecbis_liberia") {
+      dimension = "apps"
+      applicationIdSuffix = ".ecbis_liberia"
+      versionNameSuffix = "-ecbis_liberia"
       manifestPlaceholders["appLabel"] = "MOH eCBIS"
```
Comment (pld) [inline android/quest/build.gradle.kts:181]: ah so we're going to make the current one preview and have the new one be production? I think we can put "production" in the name instead of "liberia" for clarity

Comment (ndegwamartin) [reply android/quest/build.gradle.kts:181]: I think we should do the reverse since the production app is what the client will interact with even on _Playstore_ and should be concise, that is 
- For prod - the flavor should be `ecbis` and the suffix `.ecbis`
- For preview - the flavor should be `ecbis_preview` and the suffix `.ecbis_preview`

Comment (ndegwamartin) [reply android/quest/build.gradle.kts:181]: If we are unable to enforce the integrity for a fresh install then I suppose we can follow that approach. We will however need a new _Playstore_ release APK upload and verification process for the new application id cc @pld

Comment (dubdabasoduba) [reply android/quest/build.gradle.kts]: That's a nice catch. Does the mean we need to de-list the current app?

Comment (pld) [reply android/quest/build.gradle.kts]: k so then we'd have 2 releases on play store the prod and preview one? I'm fine w/that, I'd worry a bit that if we have to de-list the current app, we take the risk of delay on getting up a new app

File: android/quest/src/ecbis_preview/res/drawable/ic_app_logo.png
Comment (pld) [inline android/quest/src/ecbis_preview/res/drawable/ic_app_logo.png:1]: don't we need a different icon? like the eCBIS icon, but w/"preview" on it

Comment (pld) [inline android/quest/src/ecbis_preview/res/drawable/ic_app_logo.png:1]: shouldn't we use the ecbis logo w/a banner over it or something

Comment (dubdabasoduba) [reply android/quest/src/ecbis_preview/res/drawable/ic_app_logo.png:1]: We wanted the distinction on the installation hence the need to use different icons. 

I will be updating the configs on preview to change the app name on the UI to read preview too

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/login/LoginViewModel.kt
Comment (pld) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/login/LoginViewModel.kt:1]: what are these changes related to?

Comment (dubdabasoduba) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/login/LoginViewModel.kt:1]: These are due to some file conflicts. I think they were caused by formatting.

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/login/LoginViewModel.kt [hunk 2]
```diff
@@ -362,6 +362,7 @@ constructor(
           }
         }
       }
+
```
Comment (pld) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/login/LoginViewModel.kt:365]:
```suggestion
```

**Review by pld — APPROVED** [review]

**Review by pld — APPROVED** [review]

**Comment by ellykits** [comment]
@dubdabasoduba Do we still need to merge this PR?

**Comment by dubdabasoduba** [comment]
Closing this since it's longer needed
